extensions/cms/src/cmsUtils.ts

```diff
@@ -0,0 +1,202 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+import * as nls from 'vscode-nls';
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as mssql from '../../mssql/src/api/mssqlapis';
+import * as Utils from './cmsResource/utils';
+import { ICmsResourceNodeInfo } from './cmsResource/tree/baseTreeNodes';
+
+const localize = nls.loadMessageBundle();
+const cmsProvider: string = 'MSSQL-CMS';
+const mssqlProvider: string = 'MSSQL';
+
+/**
+ * Wrapper class to act as a facade over VSCode and Data APIs and allow us to test / mock callbacks into
+ * this API from our code
+ *
+ * @export
+ * ApiWrapper
+ */
+export class CmsUtils {
+
+	private _cmsService: mssql.CmsService;
+	private _registeredCmsServers: ICmsResourceNodeInfo[];
+
+	public showErrorMessage(message: string, ...items: string[]): Thenable<string | undefined> {
+		return vscode.window.showErrorMessage(message, ...items);
+	}
+
+	/**
+	 * Get the configuration for a extensionName
+	 * @param extensionName The string name of the extension to get the configuration for
+	 * @param resource The optional URI, as a URI object or a string, to use to get resource-scoped configurations
+	 */
+	public getConfiguration(): vscode.WorkspaceConfiguration {
+		return vscode.workspace.getConfiguration('centralManagementServers');
```

Comment:
Wait - why is this duplicated here? It's already in the API wrapper

Reply:
Some functions are needed to test extensions.